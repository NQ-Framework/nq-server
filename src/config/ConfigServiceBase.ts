export class ConfigServiceBase {
    protected throwError(param: string): never {
        throw new Error(`missing configuration item: ${param}`);
    }
}