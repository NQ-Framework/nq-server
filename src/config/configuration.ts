export default () => ({
    port: parseInt(process.env.PORT, 10) || 80,
    auth: {
        clientEmail: process.env.CLIENT_EMAIL,
        privateKey: process.env.PRIVATE_KEY,
        projectId: process.env.PROJECT_ID
    },
    db: {
    }
});