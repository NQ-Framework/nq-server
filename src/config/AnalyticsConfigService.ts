import { Injectable } from '@nestjs/common';
import { ConfigService } from '@nestjs/config';
import { ConfigServiceBase } from './ConfigServiceBase';

@Injectable()
export class AnalyticsConfigService extends ConfigServiceBase {
  constructor(private config: ConfigService) {
    super();
  }

  get measurementId(): string {
    return (
      this.config.get('analytics.measurementId') ??
      this.throwError('measurement id')
    );
  }
  get apiSecret(): string {
    return (
      this.config.get('analytics.apiSecret') ?? this.throwError('api secret')
    );
  }
}
