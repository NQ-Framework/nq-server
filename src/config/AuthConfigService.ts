import { Injectable } from '@nestjs/common';
import { ConfigService } from '@nestjs/config';
import { ConfigServiceBase } from './ConfigServiceBase';

@Injectable()
export class AuthConfigService extends ConfigServiceBase {
  constructor(private config: ConfigService) { super(); }

  get projectId(): string {
    return this.config.get('auth.projectId') ?? this.throwError('project id');
  }
  get privateKey(): string {
    return this.config.get('auth.privateKey') ?? this.throwError('private key');
  }
  get clientEmail(): string {
    return this.config.get('auth.clientEmail') ?? this.throwError('cl email');
  }
}
