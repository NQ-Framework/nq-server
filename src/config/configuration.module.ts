import { Module } from '@nestjs/common';
import { AnalyticsConfigService } from './AnalyticsConfigService';
import { AuthConfigService } from './AuthConfigService';

@Module({
  providers: [AuthConfigService, AnalyticsConfigService],
  exports: [AuthConfigService, AnalyticsConfigService],
})
export class ConfiugrationModule {}
