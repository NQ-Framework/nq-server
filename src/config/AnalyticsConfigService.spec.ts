import { Test, TestingModule } from '@nestjs/testing';
import { AnalyticsConfigService } from './AnalyticsConfigService';
import { ConfigService } from '@nestjs/config';

describe('AnalyticsConfigService', () => {
  let service: AnalyticsConfigService;

  const mockConfigService = {
    get: jest.fn(),
  };

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [
        AnalyticsConfigService,
        { provide: ConfigService, useValue: mockConfigService },
      ],
    }).compile();

    service = module.get<AnalyticsConfigService>(AnalyticsConfigService);
  });

  it('should be defined', () => {
    expect(service).toBeDefined();
  });
  it('should get settings from config service', () => {
    mockConfigService.get.mockReturnValue('test value');
    expect(service.measurementId).toEqual('test value');
    expect(service.apiSecret).toEqual('test value');
  })

  it('should throw on missing configuration', () => {
    mockConfigService.get.mockReturnValue(undefined);
    expect(() => {
      service.measurementId;
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      service.apiSecret;
    }).toThrowErrorMatchingSnapshot();
  });
});
