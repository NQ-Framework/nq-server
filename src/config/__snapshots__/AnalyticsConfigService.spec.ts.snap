// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalyticsConfigService should throw on missing configuration 1`] = `"missing configuration item: measurement id"`;

exports[`AnalyticsConfigService should throw on missing configuration 2`] = `"missing configuration item: api secret"`;
